.github/workflows/publish.yml

```diff
@@ -0,0 +1,94 @@
+# Publishes built jars to distribution platforms
+name: Publish
+
+on:
+  push:
+    tags:
+      - 'v*.*.*' # any semver tag, e.g. v1.2.3
+
+# if a second commit is pushed quickly after the first, cancel the first one's build
+concurrency:
+  group: publish-${{github.head_ref}}
+  cancel-in-progress: true
+
+env:
+  # link to the changelog with a format code for the version
+  CHANGELOG_LOCATION: "Changelog is available [here](https://github.com/${{github.repository}}/releases/tag/${{github.ref_name}})"
+  # type of release
+  RELEASE_TYPE: "beta"
+  # mark the environment as CI/CD for enabling required tasks
+  CI_CD_ENVIRONMENT: true
+
+jobs:
+  Publish:
+    runs-on: ubuntu-latest
+
+    permissions: 
+      contents: write # needed to create GitHub releases
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      
+      - name: Check for Duplicate Tags
+        run: |
+          if git rev-parse -q --verify "refs/tags/${{ github.ref }}" >/dev/null; then
+            echo "Tag already exists. A version bump is required."
+            exit 1
+          fi
+
+      - name: Setup Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '8'
+      
+      - name: Build Project
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: 'build --build-cache --daemon' # use the daemon here so the rest of the process is faster
+          generate-job-summary: false
+          gradle-home-cache-includes: |
+            caches
+            jdks
+            notifications
+            wrapper
+      
+      - name: Publish to GitHub
+        uses: softprops/action-gh-release@v1
+        with:
+          files: "build/libs/*.jar"
+          generate_release_notes: true
+          fail_on_unmatched_files: true
+
+      - name: Publish to Curseforge
+        uses: gradle/gradle-build-action@v2
+        env:
+          CURSEFORGE_API_KEY: "${{secrets.CURSEFORGE_API_KEY}}"
+          CURSEFORGE_PROJECT_ID: "${{secrets.CURSEFORGE_PROJECT_ID}}"
```

Comment:
Are secrets repository specific or organization specific? As we will have different IDs through all projects.

Reply:
They are per-repository.